lib/CMakeLists.txt

```diff
@@ -163,3 +165,12 @@ source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR}
 if(WIN32)
   set_property(DIRECTORY PROPERTY VS_STARTUP_PROJECT ${LIB_TARGET_NAME})
 endif()
+
+# Install target
+configure_package_config_file("${CMAKE_SOURCE_DIR}/cmake/config.cmake.in" "${CMAKE_BINARY_DIR}/cmake/${LIB_TARGET_NAME}Config.cmake" INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${LIB_TARGET_NAME}")
+write_basic_package_version_file("${CMAKE_BINARY_DIR}/cmake/${LIB_TARGET_NAME}ConfigVersion.cmake" VERSION "${PROJECT_VERSION}" COMPATIBILITY AnyNewerVersion)
+install(TARGETS ${LIB_TARGET_NAME} EXPORT "${LIB_TARGET_NAME}Targets" LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/")
+install(EXPORT "${LIB_TARGET_NAME}Targets" FILE "${LIB_TARGET_NAME}Targets.cmake" NAMESPACE ${PROJECT_NAMESPACE}:: DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${LIB_TARGET_NAME}")
```

Comment:
I can't see the project name here, but the suggested usage in the vcpkg port:
~~~
    find_package(Qlementine CONFIG REQUIRED)
    target_link_libraries(main PRIVATE oclero::qlementine)
~~~
Usage is easier when people have to remember just one name, not three. E.g.
~~~
    find_package(Qlementine CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qlementine::Qlementine)
~~~
(The namespace can be changed with `install(EXPORT ...)`. The exported target name can be changed with the `EXPORT_NAME` property.)

Reply:
I was thinking about swapping the `oclero::` namespace by `qlementine` for v1.1.0, so I agree with that change 👍.

Also, I plan to totally remove the `oclero::` namespace in C++, in order to just keep `qlementine::`, but that's for a future release (maybe 1.2.0 ?).

Reply:
I will fix the CMake namespace now.

Reply:
@oclero Would you like me to fix the C++ namespaces now and move the files out of the `oclero` folder too? I can do that in this PR too now...

Reply:
Okay. For CMake, it is possible to add an alternative target name already now, as (imported) interface or alias target in the config file or as exported interface target. So you can allow for a deprecation/transition period.

Reply:
> So you can allow for a deprecation/transition period.

I don't think we need this since the CMake install target has never existed before this PR.

Reply:
@nlogozzo Can it be another PR, please? Since lots of files will be removed. Thanks!

Reply:
> > So you can allow for a deprecation/transition period.
> 
> I don't think we need this since the CMake install target has never existed before this PR.

I see. Fine for the *installed* config.

But there is
~~~cmake
add_library(${PROJECT_NAMESPACE}::${LIB_TARGET_NAME} ALIAS ${LIB_TARGET_NAME})
~~~
for sub-project usage. This is what will eventually need to migrate, for consistent usage.

Reply:
Also, could you format a bit the cmake code so it's easier to read, please?
(line breaks and all)

Reply:
Another suggestion: there is still the lib's name in the `config.cmake.in`. Could you replace it by a variable, please?

Something like the following:

```cmake
check_required_components(@LIB_TARGET_NAME@)
```

Reply:
Sure.

Going to replace `LIB_TARGET_NAME` with `PROJECT_NAME` however since they are just the same no need for an extra variable.

Reply:
Okay committed all changes.

Will be creating a similar PR for `qlementine-icons` btw as I would like to add that to vcpk as well.

Reply:
Yes, the alias even becomes unnecessary if we remove the `oclero` CMake namespace anyway.